.github/workflows/release.yaml

```
@@ -26,9 +33,56 @@ jobs:
       - name: Build binary
         run: make single-binary
 
-      - name: Release
+      - name: Set Version
+        run: |
+          # set version based on dispatch type
+          if [[ "${{github.event_name}}" == "workflow_dispatch" ]]; then
+            echo "VERSION=${{ github.event.inputs.version }}" >> $GITHUB_ENV
+          elif [[ "${{github.ref_type}} == "tag" ]]; then
+            echo "VERSION=${{  github.ref_name }}" >> $GITHUB_ENV
+          else
+            echo "VERSION=latest" >> $GITHUB_ENV
+          fi
+
+      - name: Test and make manifests
+        id: test-and-manifests
+        run: |
+          # run make test to test and create manifests
+          echo "PRIMAZA_CONFIG_FILE=primaza_main_config_${{ env.VERSION }}.yaml" >> $GITHUB_ENV
+          echo "WORKER_CONFIG_FILE=primaza_worker_config_${{ env.VERSION }}.yaml" >> $GITHUB_ENV
+          echo "APPLICATION_AGENT_CONFIG_FILE=application_agent_config_${{ env.VERSION }}.yaml" >> $GITHUB_ENV
+          echo "SERVICE_AGENT_CONFIG_FILE=service_agent_config_${{ env.VERSION }}.yaml" >> $GITHUB_ENV
+          make test
+
+      - name: Delete previous release and tag
+        if: ${{ env.VERSION == 'latest' }}
+        uses: dev-drprasad/delete-tag-and-release@v0.2.0
+        with:
+          delete_release: true # default: false
+          tag_name: ${{ env.VERSION }} # tag name to delete
+
+      - name: Release for tag event
+        if: ${{ github.ref_type == 'tag' }}
         uses: softprops/action-gh-release@v1
         with:
+          body: "Release version ${{ env.VERSION }}"
```

I'd propose to add here `draft: true` and `generate_release_notes: true`.
This way we will create the Release as draft with generated releases notes. We can then update the description and manually publish the release. WDYT?

```suggestion
          body: "Release version ${{ env.VERSION }}"
          draft: true
          generate_release_notes: true
```

---

OK, did that and made the tag version the same.
As we made it draft I also removed to build on push to main - may forget it is created ad it would remove the previous release.

---

in this case I think we should also remove the job "Release for non tag events"

---

That is for the workflow dispatch which I see as the main way to create a latest tag and release.

---

you're right! LGTM